### ISSUE TITLE
Changing security config should cancel existing sessions

### ISSUE THREAD

**Issue by chillu** [Firesphere/silverstripe-bootstrapmfa#52]
# Acceptance Criteria

 * All "remember me" tokens are expired when a password is changed or reset (retaining the current browser session)
 * All "remember me" tokens are expired when a second factor is added or removed (retaining the current browser session)
 * If the system is configured for multiple "remember me" tokens, they're all expired

# Notes

From Olivier:

> Changing security configuration (like adding/removing 2nd factor or changing/resetting password) should result in cancelling all existing sessions. rational: users change their password because they think it might have been compromised.

**Comment by ScopeyNZ** [comment]
I think it probably should. I noted on Slack but this isn't necessarily a common flow. If there's the ability to log yourself out on other devices (a button labelled "log me out elsewhere" or words to that effect) then services don't always log you out of all your devices when changing your password.

Might need to have a little discussion about what suits SilverStripe best...

**Comment by robbieaverill** [comment]
@chillu are you happy for us to move this to a framework issue?

**Comment by robbieaverill** [comment]
Yeah assuming https://github.com/silverstripe/silverstripe-framework/issues/8694 gets implemented, it'd probably have an extension hook in it somewhere. Fair enough for reopening, let's re-target this issue as "ensure MFA changes are included in resetting 'remember me' cookies assuming https://github.com/silverstripe/silverstripe-framework/issues/8694 is implemented" or something to that end
